packages/jest-preset-mc-app/setup-tests.js

```diff
@@ -10,18 +10,27 @@ global.window.app = {
 
 const shouldSilenceWarnings = (...messages) =>
   [
-    /Warning: componentWillReceiveProps has been renamed/,
-    /CellMeasurerCache should only measure a cell's width or height/,
+    /.*Warning: componentWillReceiveProps has been renamed.*/,
+    /.*CellMeasurerCache should only measure a cell's width or height.*/,
   ].some(msgRegex => messages.some(msg => msgRegex.test(msg)));
 
+const shouldNotThrowWarnings = (...messages) =>
+  [/.*@commercetools-frontend\/permissions.*/].some(msgRegex =>
+    messages.some(msg => msgRegex.test(msg))
+  );
+
 // setup file
 const logOrThrow = (log, method, messages) => {
   const warning = `console.${method} calls not allowed in tests`;
   if (process.env.CI) {
-    if (shouldSilenceWarnings(messages)) {
-      return;
-    }
+    if (shouldSilenceWarnings(messages)) return;
+
     log(warning, '\n', ...messages);
+
+    // NOTE: That some warnings should be logged allowing us to refactor graceully
+    // without having to introduce a breaking change.
+    if (shouldNotThrowWarnings(messages)) return;
```

Comment:
What do you think of this @emmenko and @adnasa.

Reasoning being: we want to allow graceful migration (yes, at the cost of slowing things down). However, we promote those warnigns after being logged to errors. So in a non-breaking release we can always whitelist logs. Then say in the next major we will promote those if we can.

Actually in this case the `shouldMatchSomePermissions` will be very hard cause it's wired into the menu and all.

Reply:
Good idea!

Reply:
very nice!